thunder/transforms/quantization.py

```diff
@@ -294,3 +296,221 @@ def transform_traces_pre_prologue(self, prologue_trace, computation_trace, epilo
+                            output=None,
+                        )
+                    )
+                    new_bsyms.append(
+                        prims.check_tensor_shape_and_metadata.bind(
+                            proxy_lora_b,
+                            qs["lora_b.shape"],
+                            device,
+                            qs["lora_b.dtype"],
+                            False,
+                            output=None,
+                        )
+                    )
+                    # this is not good, because we will have several traces...
+                    additional_proxies[n_lora_a] = proxy_lora_a
+                    additional_proxies[n_lora_b] = proxy_lora_b
+
+        prologue_trace.bound_symbols[-1:-1] = new_bsyms
+        prologue_trace.set_provenance(thunder.core.trace.TraceProvenance("lora linear pass"))
+
+        computation_proxy_map = {
+            csym.name: dict(
+                shape=psym.shape,
+                dtype=psym.dtype,
+            )
+            for psym, csym in zip(prologue_trace.bound_symbols[-1].args[0][0], computation_trace.args)
+            if psym.shape != csym.shape or psym.dtype != csym.dtype
+        }
+
+        new_computation_trace = trace_with_replaced_proxy_metadata(computation_trace, computation_proxy_map)
+        bound_symbols = new_computation_trace.bound_symbols
+        new_computation_trace.bound_symbols = []
+
+        new_computation_trace.args = (*new_computation_trace.args, *new_compute_inputs)
+        new_computation_trace._siginfo.args = [(a.name, None) for a in new_computation_trace.args]
+
+        with tracectx(new_computation_trace):
+            new_bindings = [
+                thunder.core.prims.unpack_trivial.bind(i, output=i, name=i.name) for i in new_compute_inputs
+            ]
+
+        for idx, bsym in enumerate(bound_symbols):
+            if bsym.sym != prims.unpack_trivial:
+                break
+            new_computation_trace.bound_symbols.append(bsym.from_bsym())
+        new_computation_trace.bound_symbols += new_bindings
+
+        swap_map = dict()
+
+        for bsym in bound_symbols[idx:]:
+            if bsym.sym == thunder.torch.linear and bsym.args[1].name in lora_linear_proxies:
+                n = lora_linear_proxies[bsym.args[1].name]
+                with tracectx(new_computation_trace):
+                    new_computation_trace.push_scope(new_computation_trace.bound_symbols)
+                    lora_a_bsym = thunder.torch.transpose(additional_proxies[f"{n}.lora_a"], 1, 0)
+                    lora_b_bsym = thunder.torch.transpose(additional_proxies[f"{n}.lora_b"], 1, 0)
+                    lora_bsym_1 = thunder.torch.matmul(bsym.args[0], lora_a_bsym)
+                    lora_bsym = thunder.torch.matmul(lora_bsym_1, lora_b_bsym)
+                    original_weight = thunder.torch.linear(*bsym.args[:3])
+                    lora_sum = thunder.torch.add(original_weight, lora_bsym)
```

Comment:
It's a bit of a taste thing, but if we use
```
new_computation_trace.bound_symbols.append(thunder.torch.add.bind(original_weight, lora_bsym, output=var_original_bsym_output)
```
I think we would not need the `swap_map` and swapping.
(So the rule of thumb is "use calls to the symbols for getting a new output, use `.bind` when you need to specify the an existing output.)

Reply:
Ah thanks for explaining the rule of thumb (perhaps, the convention in Thunder?). Let me add that! However, when using thunder.torch.add, it was throwing an error:
```
E               NotImplementedError: VJP for torch.add is not implemented
```
However, I was able to resolve this by using prims.add. Please let me know if this can be an issue!